cpp/src/strings/split/split.cu

```diff
@@ -844,115 +860,19 @@ std::unique_ptr<experimental::table> split_fn( size_type strings_count,
             thrust::make_counting_iterator<size_type>(0),
             thrust::make_counting_iterator<size_type>(strings_count),
             d_indexes,
-            [tokenizer, col, columns_count, d_token_counts] __device__ (size_type idx) {
-                return tokenizer(idx,col,columns_count, d_token_counts);
+            [col, columns_count, d_tokens] __device__ (size_type idx) {
+                return d_tokens[ col + (idx * columns_count) ];
             });
         auto column = make_strings_column(indexes,stream,mr);
```

Comment:
If this factory `make_strings_column` accepts iterators, it would be helpful here. (and at many places).

Reply:
I think this is a good idea. I've added an iterator version of this factory and changed the code to call it.